spec/services/person_constructor_spec.rb

```diff
@@ -59,6 +59,26 @@
 
       expect(person.location.city).to eq 'city'
     end
+
+    it "normalizes zip codes before assigning to location" do
+      params = { zip_code: '12345-0001' }
+      expected_params = { zip_code: '12345' }
+      stub_person_finder(expected_params)
+
+      person = PersonConstructor.build(params)
+
+      expect(person.location.zip_code).to eq '12345'
+    end
+
+    it "rejects bad zip codes" do
+      params = { zip_code: '1231' }
+      expected_params = { zip_code: nil }
+      stub_person_finder(expected_params)
```

Comment:
It's not in `Person`, it's in `PersonConstructor`, which is a class that deals with `PersonWithRemoteFields` and `Location` params. I can see an argument for moving this into a `before_validation` callback on `Location`, though. I was trying to keep all param normalization in one place and avoid the callback on `Location`, based on something Joe Ferris said on an Upcase video about cleaning up params before they get to the model. I'm not sure his way is best. Just wanted to try it out...

I'm happy to use the callback on `Location` if you prefer.